tests/database/test_chaindb.py

```diff
@@ -53,9 +53,11 @@
 
 def set_empty_root(chaindb, header):
     root_hash = get_empty_root_hash(chaindb)
-    header.transaction_root = root_hash
-    header.receipt_root = root_hash
-    header.state_root = root_hash
+    return header.copy(
+        transaction_root=root_hash,
+        receipt_root=root_hash,
+        state_root=root_hash,
+    )
```

Comment:
This is super clear, I'm enjoying reading the `copy()` and `build_changeset()` updates.